charts/opentelemetry-operator/templates/deployment.yaml

```
@@ -32,9 +32,13 @@ spec:
             {{- end }}
           command:
             - /manager
-          {{- if .Values.manager.env }}
+          {{- if .Values.manager.env.enabled }}
```

Why is `env.enabled` added? I believe it should work as it used to be. If not, we may need `env: []` to be set as default value in `values.yaml`

---

Before there was a corner case where you would have ended up with duplicate `env` key.
If we just put `env: {}` and don't specify any kv we end up with an empty `var` key in the deployment.

---

You can use this line instead?

```suggestion
          {{- if or (not .Values.admissionWebhooks.enabled) .Values.manager.env }}
```

---

I'm pretty sure this will set the `ENABLE_WEBHOOKS: false` for any variable kv defined (only second condition matching)

---

Yes, if you remove another to check of `.Values.admissionWebhooks.enabled` inside this clause, but that's not what I'm proposing

---

ok, so basically checking `.Values.admissionWebhooks.enabled` twice.